### PR TITLE
Db sync explorer

### PR DESCRIPTION
**Pull request by dkijania**
Introducing db-sync-explorer service which expose db-sync database as as REST service. Added some helper endpoint to check meta information about db-sync and validate sync process (we can query sync process expressed as percentage or difference between current date and last registered block). Finally added api/v0/tx/hash/{...} endpoint which can return transaction information if exists in db sync. This can be useful for any clients for example GVC to check if registration transaction is confirmed

### CODE REVIEW

File: book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
+{
+	"info": {
+		"_postman_id": "cb8ec9f2-ba0d-4840-a1ce-b828936795f1",
+		"name": "Db Sync Explorer",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
+		"_exporter_id": "11181958"
+	},
```
Comment (kukkok3) [inline book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json:7]: Is _exporter_id and _postman_id starting with underscore something wanted?

File: book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+								{
+									"key": "hash",
+									"value": "bc692d5a2a6b014b50fc6216e544c6dc08299add60b62a64eb682c53ffedd4c9",
+									"description": "valid hash for given transaction"
+								}
+							]
+						}
+					},
+					"_postman_previewlanguage": null,
+					"header": null,
```
Comment (kukkok3) [inline book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json:106]: same as above

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+pub async fn get_meta_information(
+    State(pool): State<ConnectionPool>,
+) -> Result<String, (StatusCode, String)> {
+    let mut conn = pool.get().await.map_err(internal_error)?;
+
+  //  use crate::db::schema::meta;
+
+  //  let meta = meta::table.load::<Meta>(&mut *conn).map_err(internal_error)?;
+
```
Comment (kukkok3) [inline src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:24]: Some  leftover?

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        .map_err(internal_error)?;
+
+    let two: i32 = row.try_get(0).map_err(internal_error)?;
+
+    Ok(two.to_string())
+  /*  if meta.is_empty() || meta.len() > 2 {
+        Err((StatusCode::INTERNAL_SERVER_ERROR,"data inconsistency.. expected only one meta record".to_string()))
+    } else {
+        Ok(meta[0].clone())
+    }*/
```
Comment (kukkok3) [inline src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:37]: Same as above

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+let results: Vec<BehindDuration> = db.exec(move |conn| behind().load(conn)).unwrap();
+let time : DateTime<Utc> = results[0].behind_by.into();
+let diff: chrono::Duration = chrono::offset::Utc::now() - time;
+println!("{}",diff);
+
+let results: Vec<TransactionConfirmationRow> = db.exec(move |conn| hash("7b27ea78e32fdb4522ad63495a0f89289663435f3904ff5d12c529a47c8a37b8").load::<TransactionConfirmationRow>(conn)).unwrap();
+let results: Vec<TransactionConfirmation> = results.into_iter().map(TransactionConfirmation::from).collect();
+println!("{results:?}");
+
+Ok(())*/
```
Comment (kukkok3) [inline src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:57]: Same as above

**Review by kukkok3 — CHANGES_REQUESTED** [review]
Just some minor problems

File: book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json (hunk 1 above)
Comment (dkijania) [reply book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json:7]: this is autogenerated file from postman, i think it's better not to touch it

File: book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json (hunk 2 above)
Comment (dkijania) [reply book/src/core-testing-doc/api/db-sync-explorer/postman_collection.json:106]: this is autogenerated file from postman, i think it's better not to touch it

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs (hunk 3 above)
Comment (dkijania) [reply src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:24]: yes, I will completely remove file. good catch!

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs (hunk 4 above)
Comment (dkijania) [reply src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:37]: yes, I will completely remove file. good catch!

File: src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs (hunk 5 above)
Comment (dkijania) [reply src/vit-testing/db-sync-explorer/src/rest/v0/handlers.rs:57]: yes, I will completely remove file. good catch!

**Review by cong-or — APPROVED** [review]
LGTM

**Pull request merged**